packages/react-native-reanimated/src/Colors.ts

```diff
@@ -349,6 +349,10 @@ export const ColorProperties = makeShareable([
   'stroke',
 ]);
 
+const NestedColorProperties = makeShareable({
+  boxShadow: ['color'],
```

Comment:
```suggestion
  boxShadow: 'color',
```
I think we can just use a simple string instead of an array since we don't know of any more complicated structures. I understand that using an array would be a more flexible solution, but let's keep it simple for now.